components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/inbound/IdentityProcessor.java

```diff
@@ -100,6 +105,25 @@ public void init(InitConfig initConfig) {
      */
     public abstract String getCallbackPath(IdentityMessageContext context);
 
+    /**
+     * Get tenant qualified callback path.
+     *
+     * @param context IdentityMessageContext
+     * @return Tenant qualified callback path
+     */
+    protected String getTenantQualifiedCallbackPath(IdentityMessageContext context) {
+
+        try {
+            if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled() && !(new URI(getCallbackPath(context)).isAbsolute()
```

Comment:
Better to move `(new URI(getCallbackPath(context)).isAbsolute())` to a private method

Reply:
Let's move `!getCallbackPath(context).startsWith("/t/")` to a private method

Reply:
We need to handle a few cases here 

- tenant qualified mode with absolute callback path
- tenant qualified mode with relative callback path without /t/
- tenant qualified mode with relative callback path with /t/ (already tenant qualified)

- non-tenant qualified mode with absolute callback path
- non-tenant qualified mode with relative callback path without /t/
- non-tenant qualified mode with relative callback path with /t/ (already tenant qualified)

Can we have a unit test to cover these cases?

Reply:
created https://github.com/wso2/product-is/issues/9070 to track this.